src/silicone/database_crunchers/rms_closest.py

```diff
@@ -189,46 +227,61 @@ def _get_iamdf_section(self, variables):
         return iamdf_section
 
 
-def _select_closest(to_search_df, target_series):
+def _select_closest(to_search_df, target_df, weighting):
     """
-    Find row in ``to_search_df`` that is closest to the target array.
+    Find model/scenario combo in ``to_search_df`` that is closest to that of the target.
 
-    Here, 'closest' is in the root-mean squared sense. In the event that multiple rows
-    are equally close, returns first row.
+    Here, 'closest' is in the root-mean squared sense. In the event that multiple model/
+    scenarios are equally close, returns first row.
 
     Parameters
     ----------
     to_search_df : :obj:`pd.DataFrame`
-        The rows of this dataframe are the candidate closest vectors
+        The model/scenario combos to search for the closest case. A timeseries.
 
-    target_series : :obj:`pd.Series`
-        The vector to which we want to be close
+    target_df : :obj:`pd.DataFrame`
+        The data to which we want to be close. A timeseries.
+
+    weighting : map{str: float}
+        Maps the variable name onto the weighting for that variable.
 
     Returns
     -------
     dict
-        Metadata of the closest row.
+        Index of the closest timeseries.
     """
-    if len(target_series.shape) != 1:
-        raise ValueError("Target array is multidimensional")
 
-    if target_series.shape[0] != to_search_df.shape[1]:
+    if target_df.shape[1] != to_search_df.shape[1]:
         raise ValueError(
             "Target array does not match the size of the searchable arrays"
         )
-
-    closeness = []
+    if any(
+        x not in target_df.index.get_level_values("variable")
+        for x in to_search_df.index.get_level_values("variable")
+    ):
+        raise ValueError("No variable overlap between target and infiller databases.")
+    if any(col not in target_df.columns for col in to_search_df.columns):
+        raise ValueError("Time values mismatch between target and infiller databases.")
+    rms = pd.Series(index=to_search_df.index)
     for label, row in to_search_df.iterrows():
```

Comment:
if you use a groupby here you can ensure things come out in the desired order e.g.

```
for (model, scenario, variable), df in to_search_df.groupby(["model", "scenario", "variable"]):
```

A refactoring like that might make things clearer?

Reply:
I can see why this would be helpful but it's more useful to have the whole multiindex value (label) for the rms index. How about this reformatting, which should make it slightly faster too? 
`target_for_var = {}
 for var in to_search_df.index.get_level_values("variable").unique():
        target_for_var[var] = target_df[
            target_df.index.get_level_values("variable") == var
        ].squeeze()
    for label, row in to_search_df.iterrows():
        # The third item in the label is the variable name.
        rms.loc[label] = (
            ((target_for_var[label[3]] - row) ** 2).mean()
        ) ** 0.5 * weighting[label[3]]`

Reply:
Ah ok. I am more worried about the `label[3]`. Something like the below might work

```python
for label, row in to_search_df.iterrows():
    # something like this will give us a dictionary we can lookup
    ids = {k: v for k, v in zip(to_search_df.index.names, label)}
    # when you need to get things you can do something like
    ids["variable"]
```

Reply:
I've done something else, since there's no need for any category except the variable here